### ISSUE TITLE
ARC should be able to postpone tasks

### ISSUE THREAD

**Issue by carlosmiranda** [zerocracy/farm#1138]
Some tasks have impediments that are not solvable in a definite timeframe. For example, there can be a bug that requires a fix in an external library that needs to be updated. In this case the options of the ticket performer can either ask `0crat` to wait, or refuse. Both are not good options, because:

1. Waiting will cause the performer's speed to go down, and block a slot on the agenda queue from other tasks that may need attention.
2. Refusing will penalize the score and cause him to be banned from the job.

Yet this is due to no fault of the performer. A better option would be to allow the ARC to postpone or suspend the task. In this case it will be on WBS, but will not be assigned (or unassigned, if it's already assigned), and cause no penalties or bans.

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by carlosmiranda** [comment]
@g4s8 what do you think?

**Comment by g4s8** [comment]
@carlosmiranda looks like real problem, but I can't understand how to put back these tasks to WBS, if ARC can postpone a task which is blocked by external library bug then ARC have to monitor this bug status manually to "unblock" origin issue. I think it may take a lot of ARC's time to do it.

**Comment by carlosmiranda** [comment]
@g4s8 best (or least worst solution) is manual unpostponement, I think. Let's implement that now and refine the mechanism later (with expiry, etc) in case it becomes a problem. What do you think?

**Comment by carlosmiranda** [comment]
@g4s8 I think we should proceed with this. The way it works now is that performers will suspend tasks indefinitely, or refuse them. That will result in even more work for the architect because he now has to find those tasks with impediments or were refused and manually reassign them.

We can add information to the tasks to indicate if they are postponed (e.g. in WBS) so that architect has a high level view of all postponed tasks.

**Comment by g4s8** [comment]
@carlosmiranda I think we can suspend some tasks if they are blocked by external issue: ARC say `@0crat suspend`, this task removed from WBS and added to `suspends.xml` where we can keep task's performer, when external issue is solved ARC can say `@0crat continue` and this job will be moved from `suspends.xml` back to WBS and performer will be automatically assigned.

@yegor256 please, have a look here, what do you think?

**Comment by carlosmiranda** [comment]
@g4s8 sounds good, I don't think it's necessary to keep the performer though. By the time task is available again, he may be full of tasks, or even resigned

**Comment by yegor256** [comment]
@g4s8 I like the idea with `suspends.xml`, but I don't think we need to remove them from`wbs.xml`. How about we just introduce a new type of impediment instead? There will be _soft_ impediments, which can be created by programmers, and there will be _hard_ impediments, which can only be created by ARC/PO. They will be created by `suspend` command. Also, let's create `continue` command as you suggested, which will remove all impediments from the task. WDYT? @carlosmiranda

**Comment by carlosmiranda** [comment]
@yegor256 sounds good

**Comment by yegor256** [comment]
@carlosmiranda the performer should not be resigned, but the election process should not take into account hard-impeeded tasks.

**Comment by 0crat** [comment]
Job [#1138](https://github.com/zerocracy/farm/issues/1138) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)
